gazebo_ros_control/src/gazebo_ros_control_plugin.cpp

```diff
@@ -0,0 +1,509 @@
+                
+                //node->get_parameter(key).as_string_array();
+
+                if (key == "joints")
+                {
+                  if (!node->has_parameter("write_op_modes"))
+                    node->declare_parameter("write_op_modes");
+                  node->set_parameter(rclcpp::Parameter("write_op_modes", val));
+                }
+              }
+              else
+              {
+                size_t index = 0;
+                for (auto yaml_node_it : yaml_node) {
+                  std::string newkey = std::to_string((index++));
+                  if (!key.empty())
+                    newkey = key + separator + newkey;
+                  //RCLCPP_ERROR(rclcpp::get_logger("load"),"prefix_seq: %s newkey: %s", prefix.c_str(), newkey.c_str());
+                  feed_yaml_to_node_rec(yaml_node_it, newkey, node, prefix);
+                }
+              }
+            }
+            //RCLCPP_ERROR(rclcpp::get_logger("load")," exit");
+          }
+        };
+        if (lc_node->get_name() != prefix)
+            return;
+      feed_yaml_to_node_rec(yaml_node, prefix, lc_node, prefix);
+    };
+    auto load_params_from_yaml = [&](rclcpp_lifecycle::LifecycleNode::SharedPtr lc_node, 
+      const std::string & yaml_config_file, const std::string& prefix)
+    {
+      if (yaml_config_file.empty()) {
+        throw std::runtime_error("yaml config file path is empty");
+      }
+
+      YAML::Node root_node = YAML::LoadFile(yaml_config_file);
+      for (auto yaml : root_node)
+      {
+        auto nodename = yaml.first.as<std::string>();
+        if (nodename == prefix)
+          load_params_from_yaml_node(lc_node, yaml.second, prefix);
+      }
+      //load_params_from_yaml_node(lc_node, root_node, prefix);
+    };
+
+    //controller_manager_->load_controller("joint_state_controller_spawner", "joint_state_controller/JointStateController");
+
+    controller_ = controller_manager_->load_controller("coffeebot_arm_effort_controller",
+      "joint_trajectory_controller/JointTrajectoryController");
+    
+    //if (parent_model_->GetJoint("coffeebot_gripper_left_finger") && parent_model_->GetJoint("coffeebot_gripper_right_finger"))
+    controller2_ = controller_manager_->load_controller("coffeebot_gripper_effort_controller",
+      "joint_trajectory_controller/JointTrajectoryController");
+
+    //feed coffeebot_controller/coffeebot_controllers.yaml into the controllers
+    std::string param_file = 
+      ament_index_cpp::get_package_share_directory("coffeebot_controller") +
+      "/params/coffeebot_controllers.yaml";
+    RCLCPP_ERROR(rclcpp::get_logger("ros_control_plugin"),"param file: %s", param_file.c_str());
+ 
+    load_params_from_yaml(controller_->get_lifecycle_node(), param_file, "coffeebot_arm_effort_controller");
+    if (controller2_)
+      load_params_from_yaml(controller2_->get_lifecycle_node(), param_file, "coffeebot_gripper_effort_controller");
```

Comment:
Custom code: I load 2 controllers and associated parameter files here. Again, this functionality should be replaced by controller_manager spawn controller functionality when it has been implemented.